File: recursion/go/factorial.go
package main

import (
	"fmt"
)

func factorial(number int) int {
	if number > 1 {
		return number * factorial(number-1)
	}
	return 1
}

func main() {
	var n int
	fmt.Print("Enter a number: ")
	fmt.Scanf("%d", &n)
	fmt.Printf("(%d)! = %d\n", n, factorial(n))
}

File: Recursion/factorial.c
#include<stdio.h>
int sum(int);
main()
{
   int x,z;
   printf("Enter the value of x:\n");
   scanf("%d",&x);
   z=sum(x);
   printf("The sum of %d numbers is %d.\n",x,z);
}
int sum(int a)
{
    int s;
    if(a==1)
        return(a);
    s=a*sum(a-1);
    return(s);
}

